crates/moonbuild/src/gen/gen_check.rs

```diff
@@ -257,24 +265,63 @@ pub fn gen_check(
     let _ = moonc_opt;
     let _ = moonbuild_opt;
     let mut dep_items = vec![];
-    for (_, pkg) in m.get_all_packages().iter() {
-        let item = pkg_to_check_item(&pkg.root_path, m.get_all_packages(), pkg, moonc_opt)?;
+
+    // --patch and --no-mi only apply over the package which specified by path
+    let mut spec_pkg_to_check = None;
+    let mut patch_path = None;
+    let mut no_mi = false;
+
+    // if pkg is specified, check that pkg and it's deps; if no pkg specified, check all pkgs
+    let pkgs_to_check = match moonbuild_opt.check_opt.as_ref() {
```

Comment:
The filter doesn't seem as clean as its counterpart in `gen_runtest`. The logic for checking should be simpler than the logic for testing.